app_dart/test/request_handlers/get_green_commits_test.dart

```diff
@@ -0,0 +1,156 @@
+
+void main() {
+  group('GetStatus', () {
+    late FakeConfig config;
+    FakeClientContext clientContext;
+    FakeKeyHelper keyHelper;
+    FakeBuildStatusService buildStatusService;
+    late RequestHandlerTester tester;
+    late GetGreenCommits handler;
+
+    late Commit commit1;
+    late Commit commit2;
+
+    late Task task1;
+    late Task task2;
+    late Task task3;
+    late Task task4;
+    late Task task5;
+
+    late Stage stage1;
+    late Stage stage2;
+    late Stage stage3;
+    late Stage stage4;
+    late Stage stage5;
+
+    Future<T?> decodeHandlerBody<T>() async {
+      final Body body = await tester.get(handler);
+      return await utf8.decoder.bind(body.serialize() as Stream<List<int>>).transform(json.decoder).single as T?;
+    }
+
+    setUp(() {
+      clientContext = FakeClientContext();
+      keyHelper = FakeKeyHelper(applicationContext: clientContext.applicationContext);
+      tester = RequestHandlerTester();
+      config = FakeConfig(keyHelperValue: keyHelper);
+      buildStatusService = FakeBuildStatusService(commitStatuses: <CommitStatus>[]);
+      handler = GetGreenCommits(
+        config,
+        datastoreProvider: (DatastoreDB db) => DatastoreService(config.db, 5),
+        buildStatusProvider: (_) => buildStatusService,
+      );
+      commit1 = Commit(
+          key: config.db.emptyKey.append(Commit, id: 'flutter/flutter/ea28a9c34dc701de891eaf74503ca4717019f829'),
+          repository: 'flutter/flutter',
+          sha: 'ea28a9c34dc701de891eaf74503ca4717019f829',
+          timestamp: 3,
+          message: 'test message 1',
+          branch: 'master');
+      commit2 = Commit(
+          key: config.db.emptyKey.append(Commit, id: 'flutter/flutter/d5b0b3c8d1c5fd89302089077ccabbcfaae045e4'),
+          repository: 'flutter/flutter',
+          sha: 'd5b0b3c8d1c5fd89302089077ccabbcfaae045e4',
+          timestamp: 1,
+          message: 'test message 2',
+          branch: 'master');
+
+      task1 = Task(
+        key: commit1.key.append(Task, id: 123),
+        commitKey: commit1.key,
+        name: 'Linux A',
+        status: Task.statusSucceeded,
+      );
+      task2 = Task(
+        key: commit1.key.append(Task, id: 456),
+        commitKey: commit1.key,
+        name: 'Windows A',
+        status: Task.statusFailed,
+      );
+      task3 = Task(
+        key: commit2.key.append(Task, id: 123),
+        commitKey: commit2.key,
+        name: 'Linux B',
+        status: Task.statusSucceeded,
+      );
+      task4 = Task(
+        key: commit2.key.append(Task, id: 456),
+        commitKey: commit2.key,
+        name: 'Windows B',
+        status: Task.statusSucceeded,
+      );
+      task5 = Task(
+        key: commit2.key.append(Task, id: 789),
+        commitKey: commit2.key,
+        name: 'Linux C',
+        status: Task.statusSucceeded,
+      );
+
+      stage1 = Stage('cocoon', commit1, [task1, task2], Task.statusSucceeded);
```

Comment:
Can we avoid using stages? We're looking to remove them soon

Reply:
Sure. I have updated validation logic in request handler to be based on individual [Task] instead of [stage]. umm In tests, I might be wrong, but I think given the current topology of accessing tasks from commitStatus: [commitStatus] -> List<stage> -> List<task> , it would be necessary to create [stage] variables as intermediate bridges to retrieve the underlying [Task] variables?

Reply:
The tests are fine then